app/server.js

```diff
@@ -3,25 +3,268 @@ const bodyParser = require('body-parser');
+	}
+	
+	// Find the user associated with the token
+	activeUser = users.find(user => user.login.username === matchingToken.username);
+	if (!activeUser) {
+		return res.status(404).send('User not found');
+	}
+	
+	// Successful request
+	return res.send(activeUser.cart);
+});
+
+/**POST cart */
+app.post('/v1/me/cart', (req, res) => {
+	
+	// Did the request include a token?
+	const requestToken = req.headers.authorization;
+	if (!requestToken) {
+		return res.status(401).send('Authentication required: Missing token');
+	}
+	
+	// Is the token valid?
+	const matchingToken = checkToken(accessTokens, requestToken);
+	if (!matchingToken) {
+		return res.status(401).send('Authentication required: Invalid token');
+	}
+	
+	// Find the user associated with the token
+	activeUser = users.find(user => user.login.username === matchingToken.username);
+	if (!activeUser) {
+		return res.status(404).send('User not found');
+	}
+
+	const { productId, quantity } = req.body;
+	// Did the request include a product?
+	if (!productId) {
+		return res.status(400).send('Please include a product to add to the cart.');
+	}
+
+	// Check for product in list of products
+	const matchingProduct = products.find(product => product.id === productId);
+	if (!matchingProduct) {
+		return res.status(404).send('Failed to locate product with the specified ID');
+	}
+
+	// Check for product already in cart
+	const matchingCartItem = checkCart(activeUser.cart, productId)
+	if (matchingCartItem) {
+		return res.status(403).send('Unable to add a product that already exists in the cart');
+	}
+
+	// Add item to cart
+	activeUser.cart = addCartItem(activeUser.cart, matchingProduct, quantity);
+	writeUsers(users);
+	return res.status(201).send('Successfully added product to cart');
+});
+
+/**PUT cart item */
+app.put('/v1/me/cart/:productId', (req, res) => {
+	
+	// Did the request include a token?
+	const requestToken = req.headers.authorization;
+	if (!requestToken) {
+		return res.status(401).send('Authentication required: Missing token');
+	}
+
+	// Is the token valid?
+	const matchingToken = checkToken(accessTokens, requestToken);
+	if (!matchingToken) {
+		return res.status(401).send('Authentication required: Invalid token');
+	}
+
+	// Find the user associated with the token
+	activeUser = users.find(user => user.login.username === matchingToken.username);
+	if (!activeUser) {
+		return res.status(404).send('User not found');
+	}
+
+	// Did the request query parameters include a productId?
+	const productId = req.params.productId;
+	if (!productId) {
+		return res.status(400).send('Missing required query parameter');
+	}
+	
+	// Did the request body include a quantity?
+	const quantity = req.body.quantity;
+	if (!quantity) {
+		return res.status(400).send('Missing quantity from request');
+	}			
+	
+	if (Number(quantity) === 0 || quantity > 1000) {
+		return res.status(403).send('Unable to update the quantity to that value');
```

Comment:
this is not the correct error code. The client sent wrong value, its 400!
```suggestion
		return res.status(400).send('Unable to update the quantity to that value');
```